marlowe-playground-client/test/BridgeTests.purs

```diff
@@ -29,6 +36,52 @@ jsonHandling = do
     error1 :: F (Array CompilationError) <- decodeFile "test/evaluation_error1.json"
     assertRight $ runExcept error1
 
+serializationTest :: TestSuite
+serializationTest =
+  test "Contract Serialization" do
+    -- A simple test that runs the Escrow contract to completion
+    let
+      ada = Token "" ""
+
+      alicePk = PK "deadbeef"
+
+      aliceAcc = AccountId (fromIntegral 0) alicePk
+
+      bobRole = Role "Bob"
+
+      const = Constant (fromIntegral 100)
+
+      choiceId = ChoiceId "choice" alicePk
+
+      valueExpr = AddValue const (SubValue const (NegValue const))
+
+      token = Token "aa" "name"
+
+      contract =
+        When
+          [ Case (Deposit aliceAcc alicePk ada valueExpr)
+              ( Let (ValueId "x") valueExpr
+                  (Pay aliceAcc (Party bobRole) ada (UseValue (ValueId "x")) Close)
+              )
+          , Case (Choice choiceId [ Bound (fromIntegral 0) (fromIntegral 1) ])
+              ( If (ChoseSomething choiceId `OrObs` (ChoiceValue choiceId const `ValueEQ` Scale (Rational (fromIntegral 1) (fromIntegral 10)) const))
+                  (Pay aliceAcc (Account aliceAcc) token (AvailableMoney aliceAcc token) Close)
+                  Close
+              )
+          , Case (Notify (AndObs (SlotIntervalStart `ValueLT` SlotIntervalEnd) TrueObs)) Close
+          ]
+          (Slot (fromIntegral 100))
+          Close
+
+      json = encodeJSON contract
+    expectedJson <- liftEffect $ FS.readTextFile UTF8 "test/contract.json"
+    let
+      rx = unsafeRegex "\\s+" (RegexFlags { global: true, ignoreCase: true, multiline: true, sticky: false, unicode: true })
+
+      expected = replace rx "" expectedJson
+    equal expected json
+    equal (Right contract) (runExcept $ decodeJSON json)
+
```

Comment:
doesn't need to be in this PR since we want to get it merged but wouldn't quickcheck round-trip encoding for all the types be good?

Reply:
Also, this is maybe a bit over the top but you could have an app in Haskell that uses quickcheck to create a load of random JSON contracts from Haskell then use these generated files in backwards round trip tests. Just lookup all json files in the generated directory and for each one decode it then encode it and check the json matches (could decode the files as Raw JSON to avoid formatting issues. This is probably big over kill and certainly not for this PR but it would be a way to ensure that Haskell and Purescript encoding always matches in `master` branch.

Reply:
Yes, I wanted to create a Golden tests (https://ro-che.info/articles/2017-12-04-golden-tests) and use same golden jsons for both PureScript and Haskell, but I wanted to finally merge this. Next week, I hope.